ui/jetbrains/src/main/java/com/neueda/jetbrains/plugin/graphdb/jetbrains/ui/datasource/metadata/tree/MutableTreeNodeProducers.java

```diff
@@ -0,0 +1,28 @@
+package com.neueda.jetbrains.plugin.graphdb.jetbrains.ui.datasource.metadata.tree;
+
+import com.intellij.icons.AllIcons;
+import com.intellij.ui.treeStructure.PatchedDefaultMutableTreeNode;
+import com.neueda.jetbrains.plugin.graphdb.jetbrains.component.datasource.state.DataSourceApi;
+import com.neueda.jetbrains.plugin.graphdb.jetbrains.ui.datasource.tree.MetadataTreeNodeModel;
+import com.neueda.jetbrains.plugin.graphdb.jetbrains.ui.datasource.tree.Neo4jTreeNodeType;
+
+import javax.swing.tree.DefaultMutableTreeNode;
+import java.util.function.Function;
+
+public final class MutableTreeNodeProducers {
+
+    public static Function<TreeNode, DefaultMutableTreeNode> function(Neo4jTreeNodeType type, DataSourceApi dataSourceApi) {
+        return treeNode -> new PatchedDefaultMutableTreeNode(
+                new MetadataTreeNodeModel(type, dataSourceApi, treeNode.getName(), AllIcons.Nodes.Function)
+        );
+    }
+
+    public static Function<TreeNode, DefaultMutableTreeNode> directory(Neo4jTreeNodeType type, DataSourceApi dataSourceApi) {
+        return treeNode -> new PatchedDefaultMutableTreeNode(
+                new MetadataTreeNodeModel(type, dataSourceApi, treeNode.getName(), AllIcons.Nodes.Package)
+        );
+    }
+
+    private MutableTreeNodeProducers() {
```

Comment:
Constructor should be in the top of the class

Reply:
changed